torch/distributed/tensor/parallel/style.py

```diff
@@ -55,6 +56,21 @@ def __init__(self) -> None:
         super().__init__(make_input_replicate_1d, make_output_tensor)
 
 
+class PairwiseSequenceParallel(ParallelStyle):
+    """
+    PairwiseParallel concatenate colwise and rowwise styles as a fixed
+    pair like what Megatron-LM(https://arxiv.org/abs/1909.08053) is doing.
+    We assume both input and output needs to a replicate DTensor.
+
+    .. warning::
+        PairwiseParallel only supports ``nn.Multihead Attention``,
+        ``nn.Transformer`` or even-number-layer MLP for now.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(make_input_replicate_1d, make_output_shard_1d)
```

Comment:
I am ok with moving converting tensor to Shard DTensor and to_local into prepare_input and prepare_out. Open for suggestion.

Reply:
Does this align with the comments above? This is converting input to replicate and output to shard, but docstring mentioned both of them needs to be replicate DTensor?

Reply:
Thanks for calling that out. This is still in prototyping, and we need to make a decision here. Let me update the PR.